src/MagnumPlugins/WebPImporter/WebPImporter.cpp

```diff
@@ -0,0 +1,155 @@
+WebPImporter::WebPImporter(PluginManager::AbstractManager& manager, const Containers::StringView& plugin): AbstractImporter{manager, plugin} {}
+
+WebPImporter::~WebPImporter() = default;
+
+ImporterFeatures WebPImporter::doFeatures() const { return ImporterFeature::OpenData; }
+
+bool WebPImporter::doIsOpened() const { return _in; }
+
+void WebPImporter::doClose() { _in = nullptr; }
+
+void WebPImporter::doOpenData(Containers::Array<char>&& data, DataFlags dataFlags) {
+    /* Because here we're copying the data and using the _in to check if file
+       is opened, having them nullptr would mean openData() would fail without
+       any error message. It's not possible to do this check on the importer
+       side, because empty file is valid in some formats (OBJ or glTF). We also
+       can't do the full import here because then doImage2D() would need to
+       copy the imported data instead anyway (and the uncompressed size is much
+       larger). This way it'll also work nicely with a future openMemory(). */
+    if(data.isEmpty()) {
+        Error{} << "Trade::WebPImporter::openData(): the file is empty";
+        return;
+    }
+
+    /* Take over the existing array or copy the data if we can't */
+    if(dataFlags & (DataFlag::Owned|DataFlag::ExternallyOwned)) {
+        _in = std::move(data);
+    } else {
+        _in = Containers::Array<char>{NoInit, data.size()};
+        Utility::copy(data, _in);
+    }
+}
+
+UnsignedInt WebPImporter::doImage2DCount() const { return 1; }
+
+Containers::Optional<ImageData2D> WebPImporter::doImage2D(UnsignedInt, UnsignedInt) {
+
+    /* Create the structure that contains the WebP image data. */
+    WebPData webp_data;
+    webp_data.bytes = reinterpret_cast<uint8_t*>(_in.data());
+    webp_data.size = _in.size();
+
+    /* Verify the file is a WebP image file */
+    bool isWebPImage = WebPGetInfo(webp_data.bytes, webp_data.size, nullptr, nullptr);
+    if(!isWebPImage) {
+        Error() << "Trade::WebPImporter::image2D(): not a webp image";
+        return {}; // Containers::NullOpt;
+    }
+
+    /* Structures for bitstream features and decoder configuration */
+    WebPDecoderConfig config;
+    WebPBitstreamFeatures bitstream = config.input;
+    CORRADE_INTERNAL_ASSERT_OUTPUT(WebPInitDecoderConfig(&config));
+
+    /* Reading the file information into config.input */
+    if(WebPGetFeatures(webp_data.bytes, webp_data.size, &bitstream) != VP8_STATUS_OK) {
+        Error{} << "The features couldn't be retrieved.";
+        return {};
+    }
+
+    /* animated files */
+    if(bitstream.format == 0) {
+        Error{} << "animated files aren't supported.";
+        return {};
+    }
+
+    /* Image size */
+    const Vector2i size(bitstream.width, bitstream.height);
+
+    /* Channel number and pixel format (always 8-bit per channel in webp images) */
+    /* lossy or lossless bitstream without alpha transparency. */
+    int channels = 3;
+    PixelFormat pixel_format = PixelFormat::RGB8Unorm;
```

Comment:
The [Corrade Coding Style](https://doc.magnum.graphics/corrade/corrade-coding-style.html) suggests naming these variables `pixelFormat` and `colorDepth`. Also check elsewhere, `webpData` and some more examples should also be renamed.